hoomd/mpcd/ReverseNonequilibriumShearFlow.h

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+/*!
+ * \file ReverseNonequilibriumShearFlow.h
+ * \brief Declaration of Reverse nonequilibrium shear flow
+ */
+
+#ifndef MPCD_REVERSE_NONEQUILIBRIUM_SHEAR_FLOW_H_
+#define MPCD_REVERSE_NONEQUILIBRIUM_SHEAR_FLOW_H_
+
+#ifdef NVCC
+#error This header cannot be compiled by nvcc
+#endif
+
+#include "hoomd/ParticleGroup.h"
+#include "hoomd/Updater.h"
+#include <pybind11/pybind11.h>
+
+namespace hoomd
+    {
+namespace mpcd
+    {
+//! Reverse nonequilibrium shear flow updater
+/*!
+ * A flow is induced by swapping velocities in x direction based on particle position in
+ * y-direction.
+ */
+class PYBIND11_EXPORT ReverseNonequilibriumShearFlow : public Updater
+    {
+    public:
+    //! Constructor
+    ReverseNonequilibriumShearFlow(std::shared_ptr<SystemDefinition> sysdef,
+                                   std::shared_ptr<Trigger> trigger,
+                                   unsigned int num_swap,
+                                   Scalar slab_width,
+                                   Scalar p_target);
+
+    //! Destructor
+    virtual ~ReverseNonequilibriumShearFlow();
+
+    //! Apply velocity swaps
+    virtual void update(uint64_t timestep);
+
+    //! Set the maximum number of swapped pairs
+    void setNswap(unsigned int new_num_swap);
+
+    //! Set the slab width
+    void setSlabWidth(Scalar slab_width);
+
+    //! Set the target velocity
+    void setTargetMomentum(Scalar p_target);
+
+    //! Get max number of swaps
+    Scalar getNswap() const
+        {
+        return m_num_swap;
+        }
+
+    //! Get slab thickness
+    Scalar getSlabWidth() const
+        {
+        return m_slab_width;
+        }
+
+    //! Get target momentum
+    Scalar getTargetMomentum() const
+        {
+        return m_p_target;
+        }
+
+    //! Get summed exchanged momentum
+    Scalar getSummedExchangedMomentum() const
+        {
+        return m_momentum_exchange;
+        }
+
+    protected:
+    std::shared_ptr<mpcd::ParticleData> m_mpcd_pdata; //!< MPCD particle data
+    unsigned int m_num_swap;                          //!< maximum number of swaps
+    Scalar m_slab_width;                              //!< thickness of slabs
+    Scalar m_momentum_exchange;                       //!< summed momentum excange between slabs
+    Scalar2 m_lo_pos;                                 //!< position of bottom slab in box
+    Scalar2 m_hi_pos;                                 //!< position of top slab in box
```

Comment:
Following the convention for the other variables, these should be named like this:
```suggestion
    Scalar2 m_pos_lo;                                 //!< position of bottom slab in box
    Scalar2 m_pos_hi;                                 //!< position of top slab in box
```